### PR TITLE
Email Protection for Windows

### PR DESCRIPTION
**Pull request by greyivy**
**Reviewer:** @GioSensation 
**Asana:** https://app.asana.com/0/0/1204507131214872/f

## Description
Implements Email Protection for the Windows browser (autofill component).

See also: https://github.com/duckduckgo/windows-browser/pull/658

## Steps to test
1. Checkout https://github.com/duckduckgo/windows-browser/pull/658, make sure `duckduckgo-autofill` submodule is up to date
2. In the menu, click **Email Protection** -> **Enable Email Protection**
3. Login as usual
4. Once logged in, ensure an initial Duck Address is set on the Email web app
5. Logout via the menu or the Email web app

### CODE REVIEW

**Comment by greyivy** [comment]
Reverted merge from `main` due to Windows using an old version of autofill (the latest causing breakage)

**Comment by greyivy** [comment]
Hey @GioSensation ! This is ready for your review when you have a moment, please.

Unfortunately I'm not able to pull from main (due to the Windows version running an old version of autofill, 6.5.1 IIRC, and breaking with the latest). I'm not sure what the best way to deal with this would be, but I'll bring it up in the parent task.

File: src/deviceApiCalls/deviceApiCalls.json [hunk 1]
```diff
@@ -47,5 +47,38 @@
...
+    "resultValidator": "./schemas/getIsLoggedIn.result.json"
+  },
+  "getUserData": {
+    "id": "getUserDataResponse",
+    "description": "Used to get Email Protection auth credentials",
+    "resultValidator": "./schemas/getUserData.result.json"
+  },
+  "getEmailProtectionCapabilities": {
+    "id": "getEmailProtectionCapabilitiesResponse",
+    "description": "",
```
Comment (GioSensation) [inline src/deviceApiCalls/deviceApiCalls.json:71]:
```suggestion
    "description": "Used by the Email web app to determine which functionality is available to the user",
```

File: src/deviceApiCalls/schemas/getEmailProtectionCapabilities.result.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "GetEmailProtectionCapabilitiesResult",
+    "type": "object",
+    "description": "",
```
Comment (GioSensation) [inline src/deviceApiCalls/schemas/getEmailProtectionCapabilities.result.json:5]:
```suggestion
    "description": "Used by the Email web app to determine which functionality is available to the user",
```

**Review by GioSensation — APPROVED** [review]
Overall this seems great. It's a pity we're spreading the Email Protection exception to this new interface, but on the other hand it's fine for this scope. My only concern is that since we're adding schemas, we're sort of enshrining these naming conventions which can be somewhat confusing. They were created when autofill was all about Email Protection so it was obvious what `userData` and `loggedIn` meant, but we're doing so much more and these names are weird, like what is `userData`? Is it the login items? Is it some user settings? Anyway, that stuff is already everywhere and each platform has slightly different names so we'll have to harmonise everything at some point and then it will be easy to also update the names here.

I'd love if @alistairjcbrown could also take a look, as he knows the details of the web app settings and capabilities.

**Comment by GioSensation** [comment]
Also, as discussed on Asana, this branch can be used for Windows while we fix the breaking changes on more recent updates.

File: src/deviceApiCalls/schemas/storeUserData.params.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "StoreUserDataParams",
+    "type": "object",
+    "description": "Used to store Email Protection auth credentials.",
+    "additionalProperties": false,
+    "properties": {
+      "token": {
+        "type": "string"
+      },
+      "username": {
```
Comment (alistairjcbrown) [inline src/deviceApiCalls/schemas/storeUserData.params.json:11]: From my previous comment -- I think this should be:
```suggestion
      "userName": {
```

**Review by alistairjcbrown — COMMENTED** [review]
One comment about a potential mismatch of `username`/`userName` and a couple of questions, but this looks great 👍

File: integration-test/extension/public/css/autofill-host-styles_chrome.css [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+input[data-ddg-autofill] {
+    /* This url is changed for Firefox at build time */
+    background-image: url('chrome-extension://__MSG_@@extension_id__/public/../img/logo-small.svg') !important;
+    background-size: auto 24px !important;
+    background-position: center right 2px !important;
+    background-repeat: no-repeat !important;
+    background-origin: content-box !important;
+}
```
Comment (alistairjcbrown) [inline integration-test/extension/public/css/autofill-host-styles_chrome.css:29]: Just checking if this file was meant to be committed as part of this branch?

**Review by alistairjcbrown — APPROVED** [review]
One question about a new file, but LGTM! :shipit:

File: integration-test/extension/public/css/autofill-host-styles_chrome.css (hunk 4 above)
Comment (greyivy) [reply integration-test/extension/public/css/autofill-host-styles_chrome.css]: Nope! Removed, thanks.

**Comment by alistairjcbrown** [comment]
@greyivy Did this PR get superseded by #323?

**Comment by greyivy** [comment]
@alistairjcbrown it did, yes! Closing.

**Pull request closed without merging**